panther_analysis_tool/detection.py

```diff
@@ -189,3 +249,597 @@ def get_module(  # pylint: disable=arguments-differ
         self, identifier: str, body: str
     ) -> ModuleType:
         return super().from_string(identifier, body, self._tmp_dir)
+
+
+# pylint: disable=too-many-instance-attributes
+class Detection:
```

Comment:
It was considerably difficult to review changes since code was moved. If it helps, I did a `git checkout` on the `rule.py` on a previous commit and diffed the two files (detection.py & rule.py) to identify the changes. Would it be too much to rename the `Rule` class to `Detection` in the `rule.py` and modify it to be more abstract? As a final step (different commit/branch?) you can move the class in `detection.py` verbatim and define the sub-class only in `rule.py`.

Reply:
Yea, sorry about that - I just did what you suggested. Hopefully that makes it a little easier -_-